docs/adr/adr-007-monitoring-and-alerting.md

```diff
@@ -0,0 +1,234 @@
+warnings are appropriate for other situations. We decided that for the email
+alert system to be "broken" it represents a failing that will lead to a user
+noticeable issue unless an engineer intervenes. Since these alerts should
+represent a broken system requiring intervention, it should be suitable to
+contact engineers out-of-hours for them as otherwise users will be
+experiencing issues.
+
+We decided that we will re-approach our alerting with this mindset and look to
+replace alerts that are vulnerable to reflecting a busy status.
+
+## Status
+
+Pending
+
+## Consequences
+
+We will be changing the approaching to monitoring and alerting across the email
+alert system of applications in order to meet this new alerting criteria.
+
+## Monitoring Email Alert API
+
+### Sidekiq workers
+
+We will configure critical alerts, that contact engineers out-of-hours, if
+individual Sidekiq jobs fail multiple times for the following workers:
+
+- [DailyDigestInitiatorWorker][]
+- [DigestEmailGenerationWorker][]
+- [ProcessContentChangeWorker][]
+- [ProcessMessageWorker][]
+- [SendEmailWorker][]
+- [WeeklyDigestInitiatorWorker][]
+
+These workers will use the database to store the amount of times we have
+attempted to perform an action. An alert will be raised if the quantity of
+failed attempts reaches a threshold (the suggestion is 5).
+
+We will contact engineers out-of-hours if either [SendEmailWorker][] or
+[DigestEmailGenerationWorker][] have not completed at least one successful run
+within a time period (suggestion is 30 minutes) despite work existing. This
+reflects a concern that we would like support engineers to be notified quickly
+if all instances of these jobs are failing and the aforementioned retry method
+may take a long time to alert in a busy period.
+
+Similarly, we will contact engineers out-of-hours if we determine that
+[DigestRuns][DigestRun] are not created within a reasonable time frame
+(suggestion is 2 hours). Creating these is a [trivial][digest-run-creation],
+but essential, operation for producing a digest and the lack of these suggests
+the system is broken whether it is busy or not.
+
+We will also monitor that runs of [RecoverLostJobsWorker][] and
+[MetricsCollectionWorker][] are succeeding at least once within a time frame
+(suggestion is 1 hour). These jobs should continue running even when the
+system is busy and the lack of them running indicates that we no longer have
+confidence the system is monitored or able to recover. Thus, an alert will be
+created that contacts engineers out-of-hours should these never succeed in a
+time frame.
+
+For Sidekiq workers that don't affect the user noticeable aspects of Email
+Alert API, such as [EmailDeletionWorker][], we won't raise critical alerts if
+these are failing. Instead we will raise a warning if any Sidekiq workers
+exhaust all their retries and are placed in the [dead set][sidekiq-dead]. This
```

Comment:
Minor: I think it's worth mentioning (at a high level) that the retries should be adjusted so they are exhausted in a short-ish period, as otherwise someone reading may wonder when exactly this alert would go off.

Reply:
Thanks for updating 👍.